docs/administration/recent_activity/recent_activity.md

```diff
@@ -49,19 +49,180 @@ This policy is also required by user having a profile to properly view their own
+* `$id` is the ID or identifier of the manipulated object. For example, the Content ID cast to string.
+* `$action` is the identifier of the performed object manipulation. For example, `create`, `update` or `delete`.
+
+The returned `CreateActivityLogStruct` is always related to the currently logged-in user.
+
+If the object you log an activity on can become unavailable (like after a `delete` action), you might want to also log the name the object has at log time to be able to display it even when the object becomes unavailable. To add this name, use `CreateActivityLogStruct::setName` before saving the log entry.
+
+#### Context group
+
+If you log several entries at once, you can group them into a context.
+A context group counts as one item in regard to `activity_logs_limit` configuration and `ActivityLogService::findGroups`'s `$limit` argument.
+
+To open a context group, use `ActivityLogService::prepareContext` which has 2 arguments:
+
+* `$source` is describing (usually through a short identifier) what is triggering the set of actions. For example, some already existing source are `web` (like for actions from the Back Office), `graphql`, `rest` and `migration`.
+* `$description` is an optional more specific contextualisation. For example, `migration` context source is associated with the migration file name in its context description.
+
+To close a context group, use `ActivityLogService::dismissContext`.
+
+In the following example, several actions are logged into one context group, even actions triggered by cascade outside the piece of code:
 
-TODO: Illustrate the use of ActivityLogService(Interface)::build & ActivityLogService(Interface)::save
+- `my_feature`
+    - `init`
+    - `create`
+    - `publish`
+    - `simulate`
+    - `complete`
+
+``` php
+[[= include_file('code_samples/recent_activity/src/Command/ActivityLogContextTestCommand.php', 56, 76) =]]
+```
+
+Context groups can't be nested.
+If a new context is prepared while a context is already grouping log entries, this new context will be ignored.
+To start a new context, make sure to previously dismiss the existing one.
+
+As the context are not really user-friendly, by default, contexts are not shown in the Back Office.
+A context group is folded below its first entry.
+
+#### List
+
+To display your log entry, if your object's PHP class isn't already covered, you'll have to:
+
+* implement `ClassNameMapperInterface` to associate the class name with an identifier,
+* eventually create a `PostActivityListLoadEvent` subscriber if you need to load the object for the template,
+* create a template to display this class log entries.
+
+You can have a template:
+
+* specific to an action on an identifier and placed in `templates/themes/<theme>/activity_log/ui/<identifier>/<action>.html.twig`
+* specific to an identifier and placed in `templates/themes/<theme>/activity_log/ui/<identifier>.html.twig`
```

Comment:
I'd swap both entries as the second one is less specific.

Reply:
Reordered in 1c47a199f8d8e38f3d0427c9fa179f678c90b718